### PR TITLE
correct bug in box_mesh.py

### PR DESCRIPTION
**Pull request by chemiskyy**
This PR corrects a bug when the dimensions of the eve is not similar in all directions. If one of the dimension is more than 2 times another one, the boundary elements are not correctly captured.

@edit : also modified Type hint for Tuple in test of mesh periodic

### CODE REVIEW

**Review by ricardo0115 — COMMENTED** [review]
A test to verify that this bug has been fixed would be ideal if possible.

File: microgen/box_mesh.py
[the remark this answers is not in the document]
Comment (chemiskyy) [reply microgen/box_mesh.py:1]: Added 3 tests that actually fail with microgen 1.1 and pass with this PR

File: tests/test_box_mesh.py [hunk 1]
```diff
@@ -134,41 +158,73 @@ def _check_triangle_on_boundary(
...
+    "box_mesh",
+    [
+        "default_box_mesh",
+        "expanded_box_mesh_x",
+        "expanded_box_mesh_y",
+        "expanded_box_mesh_z",
+    ],
+)
+def test_given_box_mesh_construct_must_find_center_corners_edges_faces_node_sets(
+    box_mesh: BoxMesh, request
```
Comment (ricardo0115) [inline tests/test_box_mesh.py:171]: `request` parameter should be typed as `FixtureRequest`

`from pytest import FixtureRequest`

File: tests/test_box_mesh.py [hunk 2]
```diff
@@ -134,41 +158,73 @@ def _check_triangle_on_boundary(
...
+    "box_mesh",
+    [
+        "default_box_mesh",
+        "expanded_box_mesh_x",
+        "expanded_box_mesh_y",
+        "expanded_box_mesh_z",
+    ],
+)
+def test_given_box_mesh_rve_property_must_build_correct_rve(
+    box_mesh: BoxMesh, request
```
Comment (ricardo0115) [inline tests/test_box_mesh.py:201]: `request` parameter should be typed as `FixtureRequest`

`from pytest import FixtureRequest`

File: tests/test_box_mesh.py [hunk 3]
```diff
@@ -134,41 +158,73 @@ def _check_triangle_on_boundary(
...
 def test_given_box_box_mesh_boundary_elements_must_find_boundary_surface_elements(
-    default_box_mesh,
+    box_mesh: BoxMesh, request
 ) -> None:
+    box_mesh_to_test, rve = request.getfixturevalue(box_mesh)
     expected_number_of_cells = 48
-    rve = _default_rve()
-    boundary, boundary_cells_index = default_box_mesh.boundary_elements(rve)
+    boundary, boundary_cells_index = box_mesh_to_test.boundary_elements(rve)
+    print(boundary)
```
Comment (ricardo0115) [inline tests/test_box_mesh.py:226]: This `print` is necessary or it could be removed?

File: tests/test_box_mesh.py [hunk 4]
```diff
@@ -134,41 +158,73 @@ def _check_triangle_on_boundary(
...
+    print(boundary_cells_index)
```
Comment (ricardo0115) [inline tests/test_box_mesh.py:227]: This `print` is necessary or it could be removed?

File: tests/test_mesh_periodic.py [hunk 5]
```diff
@@ -175,7 +175,7 @@ def octet_truss_heterogeneous(rve_unit: Rve) -> (cq.Compound, List[Phase]):
     ],
 )
 def test_octettruss_mesh_must_be_periodic(
-    shape: Union[cq.Compound, cq.Shape],
+    shape: Tuple[Union[cq.Compound, cq.Shape], List[Phase], Rve],
     request,
```
Comment (ricardo0115) [inline tests/test_mesh_periodic.py:185]: `request` parameter should be typed as `FixtureRequest`

`from pytest import FixtureRequest`

**Review by ricardo0115 — COMMENTED** [review]
Thanks for adding the corresponding tests @chemiskyy there are some minor changes to do, to finally merge to main.
* Add FixtureRequest as type of request in tests
* Remove potential debugging `print` instructions

I've fixed the pre-commit failed job #116 https://github.com/3MAH/microgen/actions/runs/8807136252

I've renamed the variable `listPeriodicPhases` to `list_periodic_phases` in `test_mesh_periodic.py` to respect snake_case convention

**Review by ricardo0115 — APPROVED** [review]
LQGTM thanks! I allowed myself to make the last minor changes (related to type hints annotation and unused prints instructions) in this PR in order to be merged in main.

**Pull request merged**